### PR TITLE
Patient Home Page redesign

### LINKED ISSUE #8808: Improving UI of Patient Details page
The Patient Details page needs to be updated to follow the new design. The current layout should be changed to look like the design shown below:

Image

### PR DESCRIPTION
**Pull request by shivankacker**
> Original Author : @Mahendar0701 

## Proposed Changes

- Fixes #8808 
- Implemented some parts of proposed design of patient details page
- Created tabs for Patient profile, Consultation history, Test sample history, Shifting history, Visit history as suggested
- Shifted add consultation blocks to right side and made it responsive

Proposed:
image

Modified page:
image

Need further suggestion on implementing ABHA card and Hospital ID Card

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
  - Enhanced routing capabilities for patient-related views, including new pages for demographics, encounters, health profiles, immunization records, sample test history, and shifting history.
  - Introduction of a notes management component for patient notes.
  - Added a new `ShiftingTable` component for displaying patient shift data.

- **Improvements**
  - Updated layout and styling for various components, improving user experience and responsiveness.
  - Enhanced localization support with new keys for multiple languages, including English, Hindi, Kannada, Malayalam, and Marathi.
  - Improved navigation within the patient registration form based on URL parameters.
  - Adjusted visual elements, including breadcrumb styling and patient information displays.
  - Enhanced the `PatientHome` component with a new tabbed interface for better organization of patient details.
  - Streamlined the `ShiftingBlock` component for optional transfer handling.
  - Improved clarity and accuracy of information displayed in the `PatientInfoCard` and `InsuranceDetails` components.
  - Updated the `Chip` component's styling for better visual consistency.
  - Improved the user management interface with better validation and error handling.

- **Bug Fixes**
  - Improved error handling and validation in user management and patient registration components.
  - Enhanced patient registration testing for better verification of details after creation and updates.

### CODE REVIEW

**Comment by shivankacker** [comment]
Todo: 

- [x] The volunter tab is shown on the left side while its not present in the center 
image

- [x] Show Shifiting details as a list?
image

- [x] Rename `Sample Test` to `Service Request`

- [ ] Move the `Request Sample test` to Actions on the right side. Its good to have page specfic actions also going into the right nav. (example `Assing to a Volunteer` should only be avilable in demography)

**Comment by shivankacker** [comment]
Moving 4. to a new issue as it would require a lot of refactor

File: src/components/Patient/models.tsx [hunk 1]
```diff
@@ -136,7 +136,7 @@ export interface PatientModel {
   is_declared_positive?: boolean;
   last_edited?: UserBareMinimum;
   created_by?: UserBareMinimum;
-  assigned_to?: { first_name?: string; username?: string; last_name?: string };
+  assigned_to?: number | null;
```
Comment (bodhish) [inline src/components/Patient/models.tsx:139]: Why do we get a number for ID from the backend 🤔

**Review by bodhish — APPROVED** [review]
LGTM

**Pull request merged**